### ISSUE TITLE
KeyError: 'client-channel'

### ISSUE THREAD

**Issue by ziadhany** [spcl/faaskeeper#24]
```
(faaskeeper) ziad@ziad:~/PycharmProjects/faaskeeper$ ./fk.py deploy service --provider aws --config config/user_config.json 
Traceback (most recent call last):
  File "/home/ziad/PycharmProjects/faaskeeper/./fk.py", line 157, in <module>
    cli()
  File "/home/ziad/.devops/bin/faaskeeper/lib/python3.10/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/ziad/.devops/bin/faaskeeper/lib/python3.10/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/ziad/.devops/bin/faaskeeper/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ziad/.devops/bin/faaskeeper/lib/python3.10/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ziad/.devops/bin/faaskeeper/lib/python3.10/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ziad/.devops/bin/faaskeeper/lib/python3.10/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/ziad/PycharmProjects/faaskeeper/./fk.py", line 40, in wrapper
    return func(*args, **kwargs)
  File "/home/ziad/PycharmProjects/faaskeeper/./fk.py", line 73, in service
    "FK_CLIENT_CHANNEL": str(config_json["client-channel"]),
KeyError: 'client-channel'

```

**Comment by mcopik** [comment]
@ziadhany Indeed, it seems that our CI system let me merge a broken commit into master :-(

Thank you for the report, I will to try fix ASAP.

**Comment by mcopik** [comment]
@ziadhany You can try the changes in #29 - this should resolve the issue. We are still missing the SQS channel - right now, it is required that the machine from which you run FK operations has a public IP and accepts incoming TCP connections - a reasonable assumption for a server but too restrictive for development.

**Comment by ziadhany** [comment]
@mcopik Yes, the issue is now resolved. But the stack still fails 
`>>  ./fk.py deploy service --provider aws --config config/user_config.json `

![image](https://user-images.githubusercontent.com/29133904/229271191-b240a2cb-6239-49d3-b035-d6a2e263930d.png)

**Comment by ziadhany** [comment]
> @ziadhany This is a common issue related to the way stacks and Serverless Framework work, and it is hard for us to completely eliminate the issue.
> 
> Can you please try to delete the stack and recreate it again? Make sure that it was erased completely - sometimes deletion will fail because there are non-empty S3 buckets.

 
```
Serverless: Operation failed!
Serverless: View the full error output: https://us-east-1.console.aws.amazon.com/cloudformation/home?region=us-east-1#/stack/detail?stackId=arn%3Aaws%3Acloudformation%3Aus-east-1%3A486021575775%3Astack%2Ffaaskeeper-dev%2F773ef9b0-d08f-11ed-8d22-0ea9cf33745f
 
 Serverless Error ----------------------------------------
 
  ServerlessError: An error occurred: dataBucket - faaskeeper-dev-data already exists.
      at /home/ziad/PycharmProjects/faaskeeper/node_modules/serverless/lib/plugins/aws/lib/monitorStack.js:143:23
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
      at async AwsDeploy.update (/home/ziad/PycharmProjects/faaskeeper/node_modules/serverless/lib/plugins/aws/lib/updateStack.js:156:5)
 

```

@mcopik  s3 faaskeeper-dev-data already exists. , I solved it by rename `service` at `aws.yml ` because we use a service name as an env variable  `service: faaskeeper-hello`
![image](https://user-images.githubusercontent.com/29133904/229292716-befabc19-7efd-445c-9e9d-669d4244892b.png)

**Comment by mcopik** [comment]
@ziadhany Glad to hear it worked! It is likely that AWS proceeded with deletion but left the non-empty bucket.

For future reference, you can pass the flag `--clean` to `./fk.py deploy` to remove the previous stage before uploading a new one - this should help with broken deployments. We use a framework plugin to clean S3 buckets before attempting deletion. I need to make a note to improve documentation on this.

**Comment by mcopik** [comment]
Fixed with #29.
